modules/index/elastic-client/src/main/scala/ch/epfl/bluebrain/nexus/commons/es/client/ElasticBaseClient.scala

```diff
@@ -0,0 +1,31 @@
+package ch.epfl.bluebrain.nexus.commons.es.client
+
+import akka.http.scaladsl.model.{HttpRequest, StatusCode}
+import cats.MonadError
+import cats.syntax.flatMap._
+import ch.epfl.bluebrain.nexus.commons.http.HttpClient.{HttpResponseSyntax, UntypedHttpClient}
+import journal.Logger
+
+/**
+  * ElasticBaseClient provides the common methods and vals used for elastic clients.
+  *
+  * @tparam F the monadic effect type
+  */
+abstract class ElasticBaseClient[F[_]](implicit
```

Comment:
do we need this separation between `ElasticBaseClient` and `ElasticClient`?

Reply:
`ElasticQueryClient` uses `ElasticBaseClient` too. So if we want to reuse the `execute` method and the val which is used in both `ElasticClient` and `ElasticQueryClient` yes.